sdk/cosmos/src/clients/database_client.rs

```diff
@@ -35,8 +35,26 @@ impl DatabaseClient {
     }
 
     /// Get the database
-    pub fn get_database(&self) -> requests::GetDatabaseBuilder<'_, '_> {
-        requests::GetDatabaseBuilder::new(self)
+    pub async fn get_database(
+        &self,
+        mut ctx: Context,
+        options: GetDatabaseOptions,
+    ) -> Result<GetDatabaseResponse, crate::Error> {
+        let mut request = self
+            .prepare_request_with_database_name(http::Method::GET)
+            .body(bytes::Bytes::new())
+            .unwrap()
+            .into();
+        options.decorate_request(&mut request)?;
+        let response = self
+            .pipeline()
+            .send(&mut ctx, &mut request)
+            .await
+            .map_err(crate::Error::PolicyError)?
+            .validate(http::StatusCode::OK)
+            .await?;
```

Comment:
This is something the pipeline should actually do, and the types of errors, which can be retried, etc. is typically handled by the pipeline. Again, we can do this in a separate PR, but I bring it up as a design point. Having every single method repeat this is burdensome and exactly what the pipeline was meant to handle. For an example reference, see .NETs: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs. Our other language SDKs use something similar.

Reply:
Are you talking about the validation? This something that depends on the specific operation being performed, so I don't think we can move this into the pipeline. I agree it's currently fairly verbose, but I'm not sure the right way to solve that is moving it into pipelines rather than moving some of this into a function somewhere.

Reply:
Some of our SDKs use a response classifier they can pass in when building the pipeline. We don't necessarily have to be the same, but - even if `validate` is kept separate here - serializing and sending the request, and deserializing and handling the response should be done in the pipeline as much as possible. We do have some clients (Key Vault does it in a couple places) that will occasionally "override" that behavior, but it's pretty rare we need to.